tests/regression/runner.py

```diff
@@ -427,6 +435,161 @@ def test_get_transactions_object(self):
 
         eval_stat(time_cost, "get transactions object")
 
+    def test_find_transactions_obj_by_tag(self):
+        logging.debug(
+            "\n================================find transactions object by tag================================"
+        )
+
+        # cmd
+        #    0. 27 trytes tag
+        #    1. 20 trytes tag
+        #    2. multiple transactions share the same 27 trytes tag
+        #    3. 30 trytes tag
+        #    4. unicode trytes tag
+        #    5. Null trytes tag
+        rand_tag_27 = gen_rand_trytes(27)
+        rand_tag_27_multi = gen_rand_trytes(27)
+        rand_tag_20 = gen_rand_trytes(20)
+        rand_tag_30 = gen_rand_trytes(30)
+        test_cases = [
+            rand_tag_27, rand_tag_27_multi, rand_tag_20, rand_tag_30, "半導體絆倒你",
+            ""
+        ]
+
+        rand_msg = gen_rand_trytes(30)
+        rand_addr = gen_rand_trytes(81)
+        transaction_response = []
+        for i in range(3):
+            post_data = {
+                "value": 0,
+                "message": rand_msg,
+                "tag": test_cases[i],
+                "address": rand_addr
+            }
+            post_data_json = json.dumps(post_data)
+
+            if i == 2:
+                transaction_response.append([
+                    API("/transaction/", post_data=post_data_json),
+                    API("/transaction/", post_data=post_data_json)
+                ])
+            else:
+                transaction_response.append(
+                    API("/transaction/", post_data=post_data_json))
+
+        for i in range(len(transaction_response)):
+            logging.debug("find transactions obj by tag i = " + str(i) +
+                          ", tx_res = " + repr(transaction_response[i]))
+
+        response = []
+        for t_case in test_cases:
+            logging.debug("testing case = " + repr(t_case))
+            response.append(API("/tag/", get_data=t_case))
+
+        for i in range(len(response)):
+            logging.debug("find transactions obj by tag i = " + str(i) +
+                          ", res = " + repr(response[i]))
+
+        for i in range(len(response)):
```

Comment:
This loop seems too lengthy and has repetitive parts. Also with nested loop inside, variable i, j, k become unrecognizable.